File: 241008/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.cpp
#include <iostream>
using namespace std;
int main() {
    int n,cnt=0;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++)
        cin>>arr[i];
    for(int k=1; k<=n; k++)
    {
        for(int i = 0, sum; i<n; i++)
        {
            sum=0;
            for(int j=i; i+k<=n && j<i+k ; j++)
            {
                sum+=arr[j];
            }
            double temp = (double)sum/(double)k;
            if(temp - (int)temp == 0)
            {
            for(int j=i; i+k<=n && j<i+k ; j++)
            {
                if(sum==0) break;
                else if(arr[j] == temp)
                {
                    cnt++;
                    break;
                }
            }
                
            }
            //cout<<k<<" "<<temp<<" "<<i<<" "<<i+k-1<<"\n";
        }
    }
    cout<<cnt;
    return 0;
}

File: 241021/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.py
n = int(input())

arr = list(map(int, input().split()))

cnt = 0


for i in range(n):
    for j in range(i,n):
        sum_ij = sum(arr[i:j+1])
        mean = sum_ij / (j-i + 1)
        exist = False
        for k in range(i,j+1):
            
            # if j - i == 1 and arr[i] == arr[j]:
            #     continue
            #print(i,j,k, mean, arr[k])
            if mean == arr[k]:
                exist = True
        if exist:
            cnt += 1


print(cnt)